tests/client_test.py

```diff
@@ -462,8 +465,25 @@ def _get_mock_exception(self):
         exception.failure = mock.Mock()
         exception.failure.errors = [error]
         exception.error = mock.Mock()
-        exception.error.trailing_metadata = self._get_trailing_metadata_fn(
-            failed=True)
+        exception.error.trailing_metadata = self._get_trailing_metadata_fn()
+        return exception
+
+    def _get_mock_transport_exception(self):
```

Comment:
While private, future contributors would probably benefit from comments describing what these are intended to do.

Reply:
I added docstrings to these methods and the below assertions as well.